Framework/include/Jet.h

```diff
@@ -5,90 +5,177 @@ class Jet
+                int         tempJetIt  = -1;
+                
+                for( int myJetIt = 0; myJetIt < Jets.size(); ++myJetIt ) {
+               
+                    TLorentzVector myJet = Jets.at(myJetIt);
+                    //Check pT matching between Jet and Electron
+                    if( std::fabs( myJet.Pt() - myElectron.Pt() ) / myElectron.Pt() > 1.0 ) continue;
+                    double jetDeltaR = myElectron.DeltaR(myJet);
+                    
+                    if( jetDeltaR < tempDeltaR ) {
+                        tempDeltaR = jetDeltaR;
+                        tempJetIt  = myJetIt;
+                    }
+                }//END of looping through jets
+                
+                if( tempDeltaR < 0.4 ) {
+                    if( std::find(removedJetsIndices.begin(), removedJetsIndices.end(), tempJetIt ) == removedJetsIndices.end() )  removedJetsIndices.push_back( tempJetIt );
+                }
+            }//END of looping through electrons
+        }//END of NElectrons if statement
+        
+        std::sort(removedJetsIndices.begin(), removedJetsIndices.end(), std::greater<>());
+        
+        for( int i = 0; i < removedJetsIndices.size(); i++ ) {
+            goodjets_.erase( goodjets_.begin() + removedJetsIndices[i] );
+            goodjets_csv_.erase( goodjets_csv_.begin() + removedJetsIndices[i] );
+        }
+        
+        for( TLorentzVector myGoodJet : goodjets_ ){
+            if( myGoodJet.Pt() > 30 && std::fabs( myGoodJet.Eta()) < etaCut ) {
+                goodjets_pt30_->push_back(myGoodJet);
+                if( myGoodJet.Pt() > 40 && std::fabs( myGoodJet.Eta() ) < etaCut ) {
+                    goodjets_pt40_->push_back(myGoodJet);
+                    if( myGoodJet.Pt() >  45 && std::fabs( myGoodJet.Eta() ) < etaCut )  {
+                        goodjets_pt45_->push_back(myGoodJet);
+                    }
                 }
             }
         }
 
-        tr.registerDerivedVar("NJets",       Jets.size());
-        tr.registerDerivedVec("Jets_pt30",   jets_pt30_);
-        tr.registerDerivedVar("NJets_pt30", (jets_pt30_==nullptr)?0:jets_pt30_->size());
-        tr.registerDerivedVec("Jets_pt45",   jets_pt45_);
-        tr.registerDerivedVar("NJets_pt45", (jets_pt45_==nullptr)?0:jets_pt45_->size());
-
-        tr.registerDerivedVar("NCleanJets",         cleanJets.size());
-        tr.registerDerivedVec("Jets_pt30_clean",    jets_pt30_clean_);
-        tr.registerDerivedVar("NJets_pt30_clean",  (jets_pt30_clean_==nullptr)?0:jets_pt30_clean_->size());
-        tr.registerDerivedVec("Jets_pt45_clean",    jets_pt45_clean_);
-        tr.registerDerivedVar("NJets_pt45_clean",  (jets_pt45_clean_==nullptr)?0:jets_pt45_clean_->size());
+        for( double goodCsvValue : goodjets_csv_ ) {
+            goodjets_CSV_->push_back( goodCsvValue );
+        }
+        
+        for( TLorentzVector myGoodJet : goodjets_ ) {
+            goodJets_->push_back( myGoodJet );
+        }
```

Comment:
Why do you need to do this? Can you just register goodjets_csv_ and goodjets_.  Looks like unneeded copying of these two variables.